src/cluster_slot_stats.c

```diff
@@ -80,9 +78,17 @@ static void addReplySlotStat(client *c, int slot) {
     addReplyArrayLen(c, 2); /* Array of size 2, where 0th index represents (int) slot,
                              * and 1st index represents (map) usage statistics. */
     addReplyLongLong(c, slot);
-    addReplyMapLen(c, 1); /* Nested map representing slot usage statistics. */
+    addReplyMapLen(c, (server.cluster_slot_stats_enabled) ? SLOT_STAT_COUNT
+                                                          : 1); /* Nested map representing slot usage statistics. */
     addReplyBulkCString(c, "key-count");
     addReplyLongLong(c, countKeysInSlot(slot));
+
+    /* Any additional metrics aside from key-count come with a performance trade-off,
+     * and are aggregated and returned based on its server config. */
+    if (server.cluster_slot_stats_enabled) {
```

Comment:
The reply schema is going to complain about the fact we are missing fields. If we only emit them when a config is enabled. I think it's OK to either 1) Hardcode them to zero if it's disabled or 2) just show whatever value is there.

Reply:
Can we make the fields optional in the reply schema?

I'd prefer we to omit fields than to return nonsense values.